src/main/java/blackjack/controller/BlackJackGameController.java

```diff
@@ -0,0 +1,94 @@
+
+        handOutCardToPlayers(blackJackGame, shufflingMachine, players);
+        handOutCardToDealer(blackJackGame, shufflingMachine, dealer);
+
+        blackJackGame.findWinner();
+        OutputView.printCardsWithSum(players.getPlayers(), dealer);
+        OutputView.printFinalResult(players.getPlayers(), dealer.getResults());
+    }
+
+    private static void handOutCardToDealer(BlackJackGame blackJackGame, ShufflingMachine shufflingMachine, Dealer dealer) {
+        while (dealer.isUnderThanBoundary(DEALER_DRAWING_BOUNDARY)) {
+            blackJackGame.handOutCardTo(shufflingMachine, dealer);
+            OutputView.printDealerReceiveOneMoreCard();
+        }
+    }
+
+    private void handOutCardToPlayers(BlackJackGame blackJackGame, ShufflingMachine shufflingMachine, Players players) {
+        for (final Player player : players.getPlayers()) {
+            handOutCardToEachPlayer(blackJackGame, shufflingMachine, player);
+        }
+    }
+
+    private void handOutCardToEachPlayer(BlackJackGame blackJackGame, ShufflingMachine shufflingMachine, Player player) {
+        String playerAnswer = inputGameCommandToGetOneMoreCard(player);
+        while (player.isUnderThanBoundary(PLAYER_BUST_BOUNDARY) && handOutCardByCommand(blackJackGame, shufflingMachine, player, playerAnswer)) {
+            playerAnswer = inputGameCommandToGetOneMoreCard(player);
+        }
+    }
+
+    private static boolean handOutCardByCommand(BlackJackGame blackJackGame, ShufflingMachine shufflingMachine, Player player, String playerAnswer) {
+        if (playerAnswer.equals(YES_COMMAND)) {
+            blackJackGame.handOutCardTo(shufflingMachine, player);
+            OutputView.printParticipantCards(player.getName(), player.getCards());
+        }
+        if (playerAnswer.equals(NO_COMMAND)) {
+            OutputView.printParticipantCards(player.getName(), player.getCards());
+            return true;
+        }
+        return false;
+    }
+
+    private BlackJackGame generateBlackJackGame() {
+        try {
+            final String inputNames = InputView.readNames();
+            return new BlackJackGame(inputNames);
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            return generateBlackJackGame();
+        }
+    }
+
+    private String inputGameCommandToGetOneMoreCard(final Player player) {
+        try {
+            final String gameCommand = InputView.readGameCommandToGetOneMoreCard(player.getName());
+            validateCorrectCommand(gameCommand);
+            return gameCommand;
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            return inputGameCommandToGetOneMoreCard(player);
+        }
+    }
+
+    private static void validateCorrectCommand(final String gameCommand) {
+        if (!(gameCommand.equals(YES_COMMAND) || gameCommand.equals(NO_COMMAND))) {
```

Comment:
gameCommand가 null로 들어올 시, NPE가 발생될 여지가 있어요.
```suggestion
        if (!(YES_COMMAND.equals(gameCommand) || gameCommand.equals(NO_COMMAND))) {
```

Reply:
와 이런 세세한 부분까지 체크해주셔서 감사합니다...!
그런데 InputView에서 validate를 통해서 공백을 체크하는데도 null이 들어올 여지가 있나요?

Reply:
InputView 가 아닌 다른곳에서 해당 메서드를 사용할 여지가 있어서요!
(현재는 없지만요) 

방어적 복사를 사용하는 이유와 크게 다르지 않을것같아요.
(다른곳에서 수정하는 로직이 없는데, 굳이 하는이유는 잘못쓸 여지가 있기때문이죠)